runtime/parachains/src/paras.rs

```diff
@@ -412,6 +447,12 @@ impl<T: Config> Module<T> {
 			return weight;
 		}
 
+		if genesis.parachain {
```

Comment:
We might want to check if `ParaLifecycles::contains(&id)` first here.

Reply:
We don't do it now, and I think that's because we have an assumption that the callers of these functions don't double-register. I think that'd be a reasonable assumption to maintain, but we should document that.